### PR TITLE
feat: convenience factories for Scala Try to StatusReply transform

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-actor/src/main/scala/akka/pattern/StatusReply.scala [hunk 1]
```diff
@@ -94,6 +94,32 @@ object StatusReply {
...
+   * Prefer the string based error response over this one when possible to avoid tightly coupled logic across
+   * actors and passing internal failure details on to callers that can not do much to handle them. [[#fromTry]]
+   * provides a convenience factory doing that for [[scala.util.Try]].
+   *
+   * For cases where types are needed to identify errors and behave differently enumerating them with a specific
+   * set of response messages may be a better alternative to encoding them as generic exceptions.
+   *
+   * Also note that Akka does not contain pre-build serializers for arbitrary exceptions.
+   */
+  def fromTryKeepException[T](aTry: Try[T]): StatusReply[T] = new StatusReply(aTry)
```
Comment (He-Pin) [inline akka-actor/src/main/scala/akka/pattern/StatusReply.scala:109]: a better name than `aTry`?

Comment (pvlugter) [reply akka-actor/src/main/scala/akka/pattern/StatusReply.scala:109]: Just calling it `status` could work, same as the Try in the StatusReply itself.

**Review by pvlugter — APPROVED** [review]

File: akka-actor/src/main/scala/akka/pattern/StatusReply.scala [hunk 2]
```diff
@@ -94,6 +94,32 @@ object StatusReply {
...
+   * Scala API: Turn a try into a status reply
+   *
+   * Prefer the string based error response over this one when possible to avoid tightly coupled logic across
+   * actors and passing internal failure details on to callers that can not do much to handle them. [[#fromTry]]
+   * provides a convenience factory doing that for [[scala.util.Try]].
+   *
+   * For cases where types are needed to identify errors and behave differently enumerating them with a specific
+   * set of response messages may be a better alternative to encoding them as generic exceptions.
+   *
+   * Also note that Akka does not contain pre-build serializers for arbitrary exceptions.
```
Comment (ennru) [inline akka-actor/src/main/scala/akka/pattern/StatusReply.scala:107]:
```suggestion
   * Also note that Akka does not contain pre-built serializers for arbitrary exceptions.
```

File: akka-actor/src/main/scala/akka/pattern/StatusReply.scala [hunk 3]
```diff
@@ -94,6 +94,32 @@ object StatusReply {
    */
   def error[T](exception: Throwable): StatusReply[T] = Error(exception)
 
+  /**
+   * Scala API: Turn a try into a status reply
```
Comment (ennru) [inline akka-actor/src/main/scala/akka/pattern/StatusReply.scala:98]:
```suggestion
   * Scala API: Turn a Try into a status reply.
```

File: akka-actor/src/main/scala/akka/pattern/StatusReply.scala [hunk 4]
```diff
@@ -94,6 +94,32 @@ object StatusReply {
...
+   *
+   * For cases where types are needed to identify errors and behave differently enumerating them with a specific
+   * set of response messages may be a better alternative to encoding them as generic exceptions.
+   *
+   * Also note that Akka does not contain pre-build serializers for arbitrary exceptions.
+   */
+  def fromTryKeepException[T](aTry: Try[T]): StatusReply[T] = new StatusReply(aTry)
+
+  /**
+   * Scala API: Turn a try into a status reply.
```
Comment (ennru) [inline akka-actor/src/main/scala/akka/pattern/StatusReply.scala:112]:
```suggestion
   * Scala API: Turn a Try into a status reply.
```

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**